internal/controllers/machineset/machineset_delete_policy.go

```diff
@@ -97,7 +97,11 @@ type sortableMachines struct {
 func (m sortableMachines) Len() int      { return len(m.machines) }
 func (m sortableMachines) Swap(i, j int) { m.machines[i], m.machines[j] = m.machines[j], m.machines[i] }
 func (m sortableMachines) Less(i, j int) bool {
-	return m.priority(m.machines[j]) < m.priority(m.machines[i]) // high to low
+	priorityI, priorityJ := m.priority(m.machines[i]), m.priority(m.machines[j])
+	if priorityI == priorityJ {
+		return m.machines[i].Name < m.machines[j].Name
```

Comment:
Add a comment above this line to explain why we're falling back to name?

Reply:
I added a comment, as you suggested.